lib/spack/spack/test/llnl/util/link_tree.py

```diff
@@ -396,3 +397,257 @@ def test_source_merge_visitor_does_deals_with_dangling_symlinks(tmp_path: pathli
 
     # The first file encountered should be listed.
     assert visitor.files == {str(tmp_path / "view" / "file"): (str(tmp_path / "dir_a"), "file")}
+
+
+# Mocks for the os.path.exists function. Implemented fairly generically to not
+# tailor too much to the implementaion of
+# is_folder_on_case_insensitive_filesystem (should be a detail)
+def _iter_path_components(path: str):
+    drive, path = os.path.splitdrive(path)
+    if path.startswith(os.sep):
+        root = os.sep
+        path = path[len(os.sep) :]
+    else:
+        root = ""
+    parts = path.split(os.sep)
+    for i in range(len(parts)):
+        yield os.path.split(drive + root + os.path.join(*parts[: i + 1]))
+
+
+def _mock_exists_case_insensitive(path: str):
+    for head, tail in _iter_path_components(path):
+        contents = [c.lower() for c in os.listdir(head)]
+        if tail.lower() not in contents:
+            return False
+    return True
+
+
+def _mock_exists_case_sensitive(path: str):
+    for head, tail in _iter_path_components(path):
+        contents = os.listdir(head)
+        if tail not in contents:
+            return False
+    return True
+
+
+def test_is_folder_on_case_sensitive_filesystem(tmp_path: pathlib.Path, monkeypatch):
```

Comment:
Don't mock what you don't own. IMO it's sufficient to unit test `SourceMergeVisitor(normalize_paths=True)`. Just check whether it registers file-file / file-dir conflicts it would not register when `SourceMergeVisitor(normalize_paths=False)`.

Reply:
Hm, but this function is used to determine when the visitor should enter this mode. Should I just not test that functionality?

Reply:
Yeah, I would just drop it. The implementation is trivial and the test is not, and won't really help if someone comes by and refactors to pathlib etc.